Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -1047,64 +1001,38 @@ void TextureMapperLayer::paintFlattened(TextureMapperPaintOptions& options)
 
 void TextureMapperLayer::paintWith3DRenderingContext(TextureMapperPaintOptions& options)
 {
-    if (options.preserves3D) {
-        paintSelfAndChildrenWithReplica(options);
-        return;
-    }
-    SetForScope scopedPreserves3D(options.preserves3D, true);
-
-    Region overlapRegion;
-    Region nonOverlapRegion;
-    ComputeOverlapRegionData data {
-        ComputeOverlapRegionMode::Union,
-        options.textureMapper.clipBounds(),
-        overlapRegion,
-        nonOverlapRegion
-    };
-    data.clipBounds.move(-options.offset);
-    computeOverlapRegions(data, options.transform, false);
-    ASSERT(nonOverlapRegion.isEmpty());
+    Vector<TextureMapperLayer*> layers;
+    collect3DSceneLayers(layers);
 
-    auto rects = overlapRegion.rects();
-    static const size_t OverlapRegionConsolidationThreshold = 4;
-    if (rects.size() > OverlapRegionConsolidationThreshold) {
-        rects.clear();
-        rects.append(overlapRegion.bounds());
-    }
+    TextureMapperLayer3DRenderingContext context;
+    context.paint(layers, [&](TextureMapperLayer* layer, const FloatPolygon& clipArea) {
+        if (!clipArea.isEmpty())
+            options.textureMapper.beginClip(layer->toSurfaceTransform(), clipArea);
 
-    IntSize maxTextureSize = options.textureMapper.maxTextureSize();
-    for (auto& rect : rects) {
-        for (int x = rect.x(); x < rect.maxX(); x += maxTextureSize.width()) {
-            for (int y = rect.y(); y < rect.maxY(); y += maxTextureSize.height()) {
-                IntRect tileRect(IntPoint(x, y), maxTextureSize);
-                tileRect.intersect(rect);
-                auto surface = options.textureMapper.acquireTextureFromPool(tileRect.size(), { BitmapTexture::Flags::SupportsAlpha, BitmapTexture::Flags::DepthBuffer });
```

Comment:
Intermediate surface is removed completely

Reply:
I mean we can remove `Flags::DepthBuffer` flag.

Reply:
I'm not sure if I understand what you mean. Patch removes these lines. Do you mean that these lines should not be removed and only Flags::DepthBuffer removed?

Reply:
We no longer use Flags::DepthBuffer flag after this change. Let's remove Flags::DepthBuffer flag too. It's OK to split a patch if you want.

Reply:
Remove Flags::DepthBuffer from were? This patch has already removed Flags::DepthBuffer usage from TextureMapperLayer.

Do you mean that we should remove DepthBuffer enumeration value from BitmapTexture completely?

Reply:
oh, now I understand. I'm very surprised that This was the only usage for that flag in whole codebase

Reply:
Maybe it's better to do follow-up patch to remove it so that in case it's needed again later on by some other code then it's easy to revert, what do you think?

Reply:
Sounds good.